core/blocktree/blocktree.go

```diff
@@ -121,18 +140,40 @@ func (bt *BlockTree) String() string {
 }
 
 // LongestPath returns the path from the root to leftmost deepest leaf in BlockTree BT
-func (bt *BlockTree) LongestPath() []*node {
+func (bt *BlockTree) LongestPath() []*Node {
 	dl := bt.DeepestLeaf()
-	var path []*node
+	var path []*Node
 	for curr := dl; ; curr = curr.parent {
-		path = append([]*node{curr}, path...)
+		path = append([]*Node{curr}, path...)
 		if curr.parent == nil {
 			return path
 		}
 	}
 }
 
+// 
+func (bt *BlockTree) SubChain(start Hash, end Hash) []*Node {
+	sn := bt.GetNode(start)
+	eh := bt.GetNode(end)
+	return sn.subChain(eh)
+}
+
 // DeepestLeaf returns leftmost deepest leaf in BlockTree BT
-func (bt *BlockTree) DeepestLeaf() *node {
+func (bt *BlockTree) DeepestLeaf() *Node {
 	return bt.leaves.DeepestLeaf()
 }
+
+// computes the slot for a block from genesis
+// helper for now, there's a better way to do this
+func (bt *BlockTree) ComputeSlotForNode(n *Node, sd uint64) uint64 {
```

Comment:
does this assume that there's one block per slot? or how does it work

Reply:
This just pulls the arrival time from the node then adds the slot duration to the genesis block time until it exceeds the block arrival time and then outputs the slot based on that.

It does assume that we start counting slots at 1 though so I'll take a look to see if that's not true